### PR TITLE
feat: < TextField /> add red asterisks(*) when required=true

### PR DESCRIPTION
**Pull request by HarshitVashisht11**
Fixes Issue #1697 

#### Basic

- [ ] Used plop (`npm run plop`) to create a new component.
- [ ] PR has description.
- [ ] New component is functional and uses Hooks.
- [x] Component is written in TypeScript.

#### Style

- [x] Styles are added to `NewComponent.modules.scss` file inside the `NewComponent` folder.
- [ ] Component uses CSS Modules.
- [x] Design is compatible with [Monday Design System](https://design.monday.com/).
- [ ] Styles are defined using [monday-ui-style](https://github.com/mondaycom/monday-ui-style) tokens
- [ ] Displays correctly in all the themes

#### Storybook

- [ ] Stories were added to `/src/components/NewComponent/__stories__/NewComponent.stories.mdx` file.
- [ ] Stories include all flows of using the component.
- [ ] Stories implemented using [vibe-storybook-components](https://github.com/mondaycom/vibe-storybook-components) components and tokens.

#### Tests

- [ ] Tests are compliant with [TESTING_README.md](TESTING_README.md) instructions.

#### Accessibility

- [ ] Component is accessible.
- [ ] Component is compatible with [Vibe Accessibility Guidelines](https://style.monday.com/?path=/docs/foundations-accessibility--page)

### CODE REVIEW

**Comment by HarshitVashisht11** [comment]
Screenshot 2023-10-24 114916

Made the following changes

File: src/components/FieldLabel/FieldLabel.module.scss [hunk 1]
```diff
@@ -18,3 +18,6 @@
   color: var(--primary-text-color);
   padding-block: 2px;
 }
+.requiredAsterisk{
+  color: red;
```
Comment (SergeyRoyt) [inline src/components/FieldLabel/FieldLabel.module.scss:22]:
```suggestion
  color: var(--negative-color);
```

File: src/components/TextField/__stories__/TextField.stories.mdx [hunk 2]
```diff
@@ -219,6 +219,22 @@ Users can insert text.
   </Story>
 </Canvas>
 
+### Input field with asterisk
```
Comment (SergeyRoyt) [inline src/components/TextField/__stories__/TextField.stories.mdx:222]:
```suggestion
### Required input field
```

**Review by SergeyRoyt — CHANGES_REQUESTED** [review]
You did a great job!
Let's just fix a few small things by our designer request
Please, also check if all the checks are green 🙏 I think there might be some lint issues - you can configure prettier to avoid them

**Comment by HarshitVashisht11** [comment]
sure i will

**Review by SergeyRoyt — APPROVED** [review]
Great job! 💪
Thanks a lot! And congrats with the first contribution to Vibe! 🥳

**Pull request merged**